core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/sync/CustomParcelFilterService.java

```diff
@@ -0,0 +1,36 @@
+package com.sequenceiq.cloudbreak.service.upgrade.sync;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.model.ClouderaManagerProduct;
+import com.sequenceiq.cloudbreak.cluster.model.ParcelInfo;
+
+@Component
+public class CustomParcelFilterService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CustomParcelFilterService.class);
+
+    public Set<ParcelInfo> filterCustomParcels(Set<ParcelInfo> activeParcels, Set<ClouderaManagerProduct> availableProducts) {
+        Set<ParcelInfo> customParcels = findCustomParcels(activeParcels, availableProducts);
+        if (customParcels.isEmpty()) {
+            return activeParcels;
+        } else {
+            LOGGER.debug("Found custom parcels during CM parcel sync: {}", customParcels);
+            return activeParcels.stream()
```

Comment:
suggestion: it would be a little more elegant to add a field `custom` to `ParcelInfo` class, we could simplify the streams